### ISSUE TITLE
[Bug]: false positive on mod detection

### ISSUE THREAD

**Issue by kreezxil** [Griefed/ServerPackCreator#332]
### What happened?

with autodiscoverableenabled=true, this mod https://github.com/AntiqueAtlasTeam/AntiqueAtlas/blob/1.12.2/src/main/java/hunternif/mc/atlas/AntiqueAtlasMod.java gets removed from a server. https://www.curseforge.com/minecraft/modpacks/pixel-newage/files/3785879

talk to ACrazyD on our BPS discord for more details.

### What did you expect to happen?

not block the universal mod

### Version

3.4.1

### Relevant log output

```shell
https://haste.zneix.eu/aninyqybok -- config with autodiscoverableenabled=true
https://haste.zneix.eu/aninyqybok -- serverpackcreator log
https://www.toptal.com/developers/hastebin/femanuvade.apache -- modinstaller log
```

### Anything else you would like to add?

Tell me!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Griefed** [comment]
Heya Kreezxil,

thanks for reporting this. Unfortunately, the link to the serverpackcreator.log is the same as it is for the config. If you still have the log available, please send it my way so I can take a look.
As for the mod, I will see what I can do.

Cheers,
Griefed

**Comment by kreezxil** [comment]
https://pastebin.com/cDahYvff

**Comment by Griefed** [comment]
Schweet, thanks!

I was also able to reproduce this issue with the modpack export from the BH discord and I believe I have identified the cause of the problem. When I'm no longer sick and have gotten back into the swing of things, a fix will be available in the next alpha release.

The cause of the issue seems to be AntiqueAtlas containing two mods in one jar, basically. The annotation manifest of AA contains entries for `modId antiqueatlas` which is not clientside, but it additionally contains entries for `modId antiqueatlasoverlay` which **is** clientside. 

So, two mods, one JAR-file. One mod is server and clientside, the other is clientside only. You can see the problem.

Cheers,
Griefed

**Comment by Griefed** [comment]
Please see whether this behaviour still occurs with 3.5.0-alpha.3: https://github.com/Griefed/ServerPackCreator/releases/tag/3.5.0-alpha.3

**Comment by kreezxil** [comment]
this works in Linux no issue now.

**Comment by Griefed** [comment]
Eeeyyyyyyyyyyy it's ya boi, false positives. He be back haunting my dreams.

**Comment by Griefed** [comment]
This time, newer Forge mods are affected. TOML scanning seems to have a problem.
A good example is the ImmersiveFX mod, which has a horribly formatted toml and gets excluded even though it shouldn't be.

**Comment by Griefed** [comment]
Next version will have a fix for this issue. dev-builds are already running and proving to no longer have this issue.

**Comment by Griefed** [comment]
Aaaaaaaaaaaaand fixed again as of [3.10.0](https://github.com/Griefed/ServerPackCreator/releases/tag/3.10.0)
